virtual_ship/instruments/ctd.py

```diff
@@ -37,77 +40,89 @@ def _sample_salinity(particle, fieldset, time):
 
 
 def _ctd_cast(particle, fieldset, time):
-    # Lowering and raising CTD
-    if (
-        -fieldset.bathymetry[time, particle.depth, particle.lat, particle.lon]
-        > particle.max_depth
-    ):
-        maxdepth = (
-            -fieldset.bathymetry[time, particle.depth, particle.lat, particle.lon] + 20
-        )
-    else:
-        maxdepth = particle.max_depth
-    winch_speed = -1.0  # sink and rise speed in m/s
-
+    # lowering
     if particle.raising == 0:
-        # Sinking with winch_speed until near seafloor
-        particle_ddepth = winch_speed * particle.dt
-        if particle.depth <= maxdepth:
+        particle_ddepth = -particle.winch_speed * particle.dt
+        if particle.depth + particle_ddepth < particle.max_depth:
             particle.raising = 1
-
-    if particle.raising == 1:
-        # Rising with winch_speed until depth is -2 m
-        if particle.depth < -2:
-            particle_ddepth = -winch_speed * particle.dt
-            if particle.depth + particle_ddepth >= -2:
-                # to break the loop ...
-                particle.state = 41
-                print("CTD cast finished.")
+            particle_ddepth = -particle_ddepth
+    # raising
+    else:
+        particle_ddepth = particle.winch_speed * particle.dt
+        if particle.depth + particle_ddepth > -particle.min_depth:
+            particle.delete()
 
 
 def simulate_ctd(
-    ctds: list[CTD],
     fieldset: FieldSet,
-    out_file_name: str,
+    out_path: str | py.path.LocalPath,
+    ctds: list[CTD],
     outputdt: timedelta,
 ) -> None:
     """
     Use parcels to simulate a set of CTDs in a fieldset.
 
-    :param ctds: A list of CTDs to simulate.
     :param fieldset: The fieldset to simulate the CTDs in.
-    :param out_file_name: The file to write the results to.
+    :param out_path: The path to write the results to.
+    :param ctds: A list of CTDs to simulate.
     :param outputdt: Interval which dictates the update frequency of file output during simulation
+    :raises ValueError: Whenever provided CTDs, fieldset, are not compatible with this function.
     """
-    lon = [ctd.spacetime.location.lon for ctd in ctds]
-    lat = [ctd.spacetime.location.lat for ctd in ctds]
-    time = [ctd.spacetime.time for ctd in ctds]
+    WINCH_SPEED = 1.0  # sink and rise speed in m/s
+    DT = 10.0  # dt of CTD simulation integrator
```

Comment:
Why is this variable in capitals? We normally use lowercase for `dt`

Reply:
it is a constant, and in PEP8 constants have to be all caps